controllers/greptimedbcluster/deployers/meta.go

```diff
@@ -129,11 +138,28 @@ func (d *MetaDeployer) CheckAndUpdateStatus(ctx context.Context, highLevelObject
 	cluster.Status.Meta.Replicas = *deployment.Spec.Replicas
 	cluster.Status.Meta.ReadyReplicas = deployment.Status.ReadyReplicas
 	cluster.Status.Meta.EtcdEndpoints = cluster.Spec.Meta.EtcdEndpoints
+
+	ready := k8sutil.IsDeploymentReady(deployment)
+
+	// It should meet the following conditions to turn on maintenance mode:
+	// 1. The cluster is in starting phase that means the cluster is in the process of being created.
+	// 2. The meta deployment is ready and the maintenance mode is not enabled.
+	// 3. The `maintenanceModeWhenCreateCluster` is true in meta deployer options.
+	if d.maintenanceModeWhenCreateCluster &&
+		cluster.Status.ClusterPhase == v1alpha1.PhaseStarting &&
+		ready && !cluster.Status.Meta.MaintenanceMode {
+		// Turn on maintenance mode for metasrv.
+		if err := common.SetMaintenanceMode(common.GetMetaHTTPServiceURL(cluster), true); err != nil {
```

Comment:
How about logging the error here?

Reply:
The error logs will be propagated by the reconciler.